src/bin/cargo/commands/clean.rs

```diff
@@ -19,223 +20,128 @@ pub fn cli() -> Command {
-        .arg(
-            opt(
-                "max-git-db-age",
-                "Deletes git dependency clones that have not been used \
-                since the given age (unstable)",
-            )
-            .value_name("DURATION")
-            .value_parser(parse_time_span)
-            .hide(true),
-        )
-        .arg(
-            opt(
-                "max-download-age",
-                "Deletes any downloaded cache data that has not been used \
-                since the given age (unstable)",
-            )
-            .value_name("DURATION")
-            .value_parser(parse_time_span)
-            .hide(true),
-        )
-        .arg(
-            opt(
-                "max-src-size",
-                "Deletes source cache files until the cache is under the given size (unstable)",
-            )
-            .value_name("SIZE")
-            .value_parser(parse_human_size)
-            .hide(true),
-        )
-        .arg(
-            opt(
-                "max-crate-size",
-                "Deletes crate cache files until the cache is under the given size (unstable)",
-            )
-            .value_name("SIZE")
-            .value_parser(parse_human_size)
-            .hide(true),
-        )
-        .arg(
-            opt(
-                "max-git-size",
-                "Deletes git dependency caches until the cache is under the given size (unstable)",
-            )
-            .value_name("SIZE")
-            .value_parser(parse_human_size)
-            .hide(true),
-        )
-        .arg(
-            opt(
-                "max-download-size",
-                "Deletes downloaded cache data until the cache is under the given size (unstable)",
-            )
-            .value_name("SIZE")
-            .value_parser(parse_human_size)
-            .hide(true),
-        )
-        // These are unimplemented. Leaving here as a guide for how this is
-        // intended to evolve. These will likely change, this is just a sketch
-        // of ideas.
-        .arg(
-            opt(
-                "max-target-age",
-                "Deletes any build artifact files that have not been used \
-                since the given age (unstable) (UNIMPLEMENTED)",
-            )
-            .value_name("DURATION")
-            .value_parser(parse_time_span)
-            .hide(true),
-        )
-        .arg(
-            opt(
-                "max-target-size",
-                "Deletes build artifact files until the cache is under the given size \
-                (unstable) (UNIMPLEMENTED)",
-            )
-            .value_name("SIZE")
-            .value_parser(parse_human_size)
-            .hide(true),
+        .args_conflicts_with_subcommands(true)
+        .subcommand(
+            subcommand("gc")
+                .about("Clean global caches")
+                .hide(true)
+                // FIXME: arg_quiet doesn't work because `config_configure`
+                // doesn't know about subcommands.
```

Comment:
Side note: what do we do about `cargo report`?  I assume this would also affect #11879?

Reply:
So far, `cargo report` doesn't have a need for the `-q` flag because its only purpose is to display something on the console, so there isn't any output that needs suppressing. But it is something we should consider moving forward.

I think it will be a problem for #11879. I don't really know the best way to fix this. I don't suppose clap has a way to say "if this arg appears anywhere"?  Perhaps another option is to have each subcommand manually handle setting quiet/verbose settings?  Do you have any ideas?

Reply:
clap has `global(true)` which propagates a flag definition down through the commands and then propagates the parsed value back up and is used by all of the flags in `cli.rs`'s `cli()`.

Reply:
I opened https://github.com/rust-lang/cargo/issues/12957 to continue this conversation. We currently don't use `global(true)` because we want to be able to customize the help text per-command.